### PR TITLE
Feature/refresh only added tenants

### PR DESCRIPTION
**Pull request by sergeysenja1992**

### CODE REVIEW

File: src/main/java/com/icthh/xm/ms/configuration/service/TenantAliasService.java [hunk 1]
```diff
@@ -49,12 +52,44 @@ public List<Configuration> processConfiguration(Configuration configuration,
...
+                    .map(this::getParentKey)
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .distinct()
+                    .forEach(memoryConfigStorage::reprocess);
+
+            allTenants.stream().filter(it -> isTenantChanged(oldTenants, newTenants, it))
+                    .distinct()
+                    .forEach(configurationService::refreshTenantConfigurations);
+
```
Comment (monassfaro) [inline src/main/java/com/icthh/xm/ms/configuration/service/TenantAliasService.java:77]: What do you think of this approach?
```suggestion
            allTenants.stream().filter(it -> isTenantChanged(oldTenants, newTenants, it))
                    .map(newTenants::get)
                    .map(this::getParentKey)
                    .filter(Optional::isPresent)
                    .map(Optional::get)
                    .distinct()
                    .peek(memoryConfigStorage::reprocess)
                    .filter(it -> isTenantChanged(oldTenants, newTenants, it))
                    .distinct()
                    .forEach(configurationService::refreshTenantConfigurations);


```

Comment (sergeysenja1992) [reply src/main/java/com/icthh/xm/ms/configuration/service/TenantAliasService.java:77]: this is not equals change

**Review by major13ua — APPROVED** [review]

File: src/main/java/com/icthh/xm/ms/configuration/service/TenantAliasService.java (hunk 1 above)
Comment (sergeysenja1992) [reply src/main/java/com/icthh/xm/ms/configuration/service/TenantAliasService.java:77]: refreshTenantConfigurations by tenants

process by parents

File: src/main/java/com/icthh/xm/ms/configuration/service/TenantAliasService.java [hunk 2]
```diff
@@ -49,12 +52,44 @@ public List<Configuration> processConfiguration(Configuration configuration,
...
         } catch (IOException e) {
             log.error("Error parse tenant alias config", e);
         }
         return Collections.emptyList();
     }
+
+    private boolean isTenantChanged(Map<String, TenantAlias> oldTenants, Map<String, TenantAlias> newTenants, String it) {
```
Comment (monassfaro) [inline src/main/java/com/icthh/xm/ms/configuration/service/TenantAliasService.java:84]: I'd rather rename "it" for a more descriptive name. At the moment, it is not entirely clear what is it - tenantName?

Comment (sergeysenja1992) [reply src/main/java/com/icthh/xm/ms/configuration/service/TenantAliasService.java]: fixed

**Pull request merged**